crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -456,5 +456,257 @@ static_assert(not is_subtype_of(Intersection[Unknown, int], int))
+
+def positional_only(a: int, /) -> None: ...
+def standard(a: int) -> None: ...
+def keyword_only(*, a: int) -> None: ...
+def variadic(*args: int) -> None: ...
+def keyword_variadic(**kwargs: int) -> None: ...
+
+static_assert(not is_subtype_of(CallableTypeFromFunction[positional_only], CallableTypeFromFunction[standard]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[positional_only], CallableTypeFromFunction[keyword_only]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[positional_only], CallableTypeFromFunction[variadic]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[positional_only], CallableTypeFromFunction[keyword_variadic]))
+```
+
+But, a positional-only parameter can be a subtype of a standard parameter:
+
+```py
+static_assert(is_subtype_of(CallableTypeFromFunction[standard], CallableTypeFromFunction[positional_only]))
+```
+
+#### Standard
+
+A standard parameter is either a positional or a keyword parameter.
+
+Unlike positional-only parameters, standard parameters should have the same name in the subtype.
+
+```py
+from knot_extensions import CallableTypeFromFunction, is_subtype_of, static_assert
+
+def int_param_a(a: int) -> None: ...
+def int_param_b(b: int) -> None: ...
+
+static_assert(not is_subtype_of(CallableTypeFromFunction[int_param_a], CallableTypeFromFunction[int_param_b]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[int_param_b], CallableTypeFromFunction[int_param_a]))
+```
+
+Apart from the name, it behaves the same as positional-only parameters.
+
+```py
+def float_param(a: float) -> None: ...
+def int_param(a: int) -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[float_param], CallableTypeFromFunction[int_param]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[int_param], CallableTypeFromFunction[float_param]))
+```
+
+With the same rules for default values as well.
+
+```py
+def float_with_default(a: float = 1) -> None: ...
+def int_with_default(a: int = 1) -> None: ...
+def empty() -> None: ...
+
+static_assert(is_subtype_of(CallableTypeFromFunction[float_with_default], CallableTypeFromFunction[int_with_default]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[int_with_default], CallableTypeFromFunction[float_with_default]))
+
+static_assert(is_subtype_of(CallableTypeFromFunction[int_with_default], CallableTypeFromFunction[int_param]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[int_param], CallableTypeFromFunction[int_with_default]))
+
+static_assert(is_subtype_of(CallableTypeFromFunction[int_with_default], CallableTypeFromFunction[empty]))
+static_assert(not is_subtype_of(CallableTypeFromFunction[empty], CallableTypeFromFunction[int_with_default]))
+```
+
+#### Standard with other kinds
+
+If the corresponding parameter in the subtype is a keyword-only parameter, it behaves in the same
+way. This is because keyword-only parameter is one of the kind of standard parameter.
```

Comment:
I think we need to expand on this slightly; "behaves in the same way" isn't very clear what it's referring to.

Also, the first part seems to be phrased backward? If you take a pair of callable types, each with a single standard parameter, that form a subtype relation, you can't replace the standard parameter with a keyword-only parameter in the _subtype_ and have it "behave the same way", you can only do that in the supertype.

Reply:
Yeah, thanks for catching that, I've updated this.